pkg/credentialprovider/azure/azure_credentials.go

```diff
@@ -301,3 +302,26 @@ func (a *acrProvider) parseACRLoginServerFromImage(image string) string {
 
 	return ""
 }
+
+// shouldUseCache decide whether to use cache according to image and cache content
+func shouldUseCache(image string, d credentialprovider.DockerConfig) bool {
+	if len(d) > 2 {
+		// for service principal(len=5), always use cache
+		return true
+	}
+
+	fields := strings.Split(image, "/")
+	if len(fields) <= 1 || !strings.Contains(fields[0], azureCRName) {
+		// not ACR image, use cache
+		return true
+	}
+
+	loginServer := fields[0]
+	for k := range d {
+		if strings.EqualFold(loginServer, k) {
```

Comment:
are the credential keys always just the server, or do they sometimes include additional port/path info?

Reply:
does this handle port defaulting correctly (e.g. no port == 443)?

Reply:
credential key is always the server, e.g. `foo.azurecr.io`, no port/path included